### PR TITLE
refactor(core): rename some file names and methods

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Rename some file names and methods, fix some comments

- Rename the old validators folder to libraries. 
- Fix some comments

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/experience/classes/libraries/provision-library.ts [hunk 1]
```diff
@@ -43,10 +43,10 @@ export class ProvisionLibrary {
   /**
    * Insert a new user into the Logto database using the provided profile.
    *
-   * - Provision the organization for the new user based on the profile
-   * - OSS only, new user provisioning
+   * - Provision all JIT organizations for the user if necessary.
+   * - Assign first user admin role and default tenant organization membership. [OSS only]
```
Comment (gao-sun) [inline packages/core/src/routes/experience/classes/libraries/provision-library.ts:47]:
```suggestion
   * - Assign the first user to the admin role and the default tenant organization membership. [OSS only]
```

File: packages/core/src/routes/experience/classes/libraries/provision-library.ts [hunk 2]
```diff
@@ -43,10 +43,10 @@ export class ProvisionLibrary {
...
    */
-  async provisionNewUser(profile: InteractionProfile) {
+  async createNewUser(profile: InteractionProfile) {
```
Comment (gao-sun) [inline packages/core/src/routes/experience/classes/libraries/provision-library.ts:49]:
```suggestion
  async createUser(profile: InteractionProfile) {
```

File: packages/core/src/routes/experience/classes/libraries/provision-library.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export class ProvisionLibrary {
     return user;
   }
 
-  async provisionNewSsoIdentity(
+  async addNewSsoIdentityToUser(
```
Comment (gao-sun) [inline packages/core/src/routes/experience/classes/libraries/provision-library.ts:92]:
```suggestion
  async addSsoIdentityToUser(
```

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/routes/experience/classes/libraries/provision-library.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/routes/experience/classes/libraries/provision-library.ts]: updated

File: packages/core/src/routes/experience/classes/libraries/provision-library.ts (hunk 3 above)
Comment (simeng-li) [reply packages/core/src/routes/experience/classes/libraries/provision-library.ts]: updated

**Pull request merged**